saturn-datepicker/src/datetime/date-adapter.ts

```diff
@@ -238,10 +242,20 @@ export abstract class DateAdapter<D> {
    * @returns 0 if the dates are equal, a number less than 0 if the first date is earlier,
    *     a number greater than 0 if the first date is later.
    */
-  compareDate(first: D, second: D): number {
-    return this.getYear(first) - this.getYear(second) ||
+  compareDate(first: D | number, second: D | number): number {
+    if (first === Infinity && second === Infinity) {
+      return 0;
```

Comment:
What if first is `-Infinity`?

Reply:
I understand that that is a possible case but I don't see why that would happen